src/test_fixture/sharing.rs

```diff
@@ -18,6 +18,21 @@ pub fn share<F: Field, R: RngCore>(input: F, rng: &mut R) -> [Replicated<F>; 3]
     ]
 }
 
+/// Shares `input` into 3 maliciously secure replicated secret shares using the provided `rng` implementation
+///
+#[allow(clippy::missing_panics_doc)]
+pub fn share_malicious<F: Field, R: RngCore>(x: F, rng: &mut R) -> [MaliciousReplicated<F>; 3] {
+    let rx = F::from(rng.gen::<u128>()) * x;
+    share(x, rng)
+        .iter()
+        .zip(share(rx, rng))
+        .map(|(x, rx)| MaliciousReplicated::new(*x, rx))
+        // TODO: each_ref when stable
```

Comment:
Does each_ref() work with zip?

Reply:
it should work if `array::zip` is stable https://github.com/rust-lang/rust/issues/80094. I'll mention both in the todo